includes/structured-data/class-structured-data.php

```diff
@@ -0,0 +1,312 @@
+		$post_object    = get_post();
+		$post_headline  = $post_object->post_title;
+		$post_published = str_replace( ' ', 'T', $post_object->post_date_gmt );
+		$post_modified  = str_replace( ' ', 'T', $post_object->post_modified_gmt );
+		$post_id        = $post_object->ID;
+		$post_permalink = get_permalink( $post_id );
+
+		// images with hard crops for required sizes.
+		$post_image_1x1  = wp_get_attachment_image_url( get_post_thumbnail_id( $post_id ), 'structured-1x1' );
+		$post_image_4x3  = wp_get_attachment_image_url( get_post_thumbnail_id( $post_id ), 'structured-4x3' );
+		$post_image_16x9 = wp_get_attachment_image_url( get_post_thumbnail_id( $post_id ), 'structured-16x9' );
+
+		// author list
+		$author_list = self::get_author_list();
+
+		$post_data = array(
+			array(
+				'@context'         => 'http://www.schema.org',
+				'@type'            => self::get_type( $post_id ),
+				'mainEntityOfPage' => array(
+					'@type' => 'Webpage',
+					'@id'   => $post_permalink,
+				),
+				'headline'         => $post_headline,
+				'image'            => array(
+					$post_image_1x1,
+					$post_image_4x3,
+					$post_image_16x9,
+				),
+				'datePublished'    => $post_published,
+				'dateModified'     => $post_modified,
+				'author'           => $author_list,
+				'publisher'        => array(
+					'@type' => 'Organization',
+					'name'  => $blog_info,
+					'logo'  => array(
+						'@type' => 'ImageObject',
+						'url'   => $icon_url,
+					),
+				),
+			),
+		);
+
+		return $post_data;
+	}
+
+	/**
+	 * Get Breadcrumb Data
+	 * 
+	 * Gets the schema for the post's breadcrumb data
+	 * 
+	 * @return array
+	 */
+	public static function get_breadcrumb_data(): array {
+		$crumbs = apply_filters( 'cata_structured_data_breadcrumbs', array(), get_the_ID() );
+	
+		if ( empty( $crumbs ) ) {
+			return array();
+		}
+
+		$breadcrumb_list = self::get_breadcrumb_list( $crumbs );
+
+		if ( empty( $breadcrumb_list ) ) {
+			return array();
+		}
+
+		return array(
+			array(
+				'@context'        => 'https://schema.org',
+				'@type'           => 'BreadcrumbList',
+				'itemListElement' => $breadcrumb_list,
+			),
+		);
+	}
+
+	/**
+	 * Get Oldest Post Date
+	 *
+	 * Gets the publish date of the oldest post
+	 *
+	 * @return string
+	 */
+	public static function get_oldest_post_date(): string {
+		$query = new \WP_Query( 
+			array(
+				'orderby'        => 'post_date',
+				'post_type'      => 'post',
+				'posts_per_page' => '1',
+				'order'          => 'ASC',
+				'no_found_rows'  => true,
+			)
+		);
+
+		if( ! $query->have_posts() ) {
+			return '';
+		}
+
+		$query->the_post();
+		$post_date = get_the_date( 'Y-m-d' );
+		wp_reset_postdata();
+		
+		return $post_date;
+	}
+
+	/** 
+	 * Get Author List
+	 * 
+	 * Generates an array of authors for the current post
+	 * 
+	 * @return array
+	 */
+	public static function get_author_list(): array {
+		// get_coauthors.
+		$coauthor_info = get_coauthors();
```

Comment:
This depends on Co-Authors Plus. We have every reason to think it will always be part of our sites but its probably still worth checking if this function exists before using it in another plugin.

Reply:
Added a check for if `get_coauthors()` exists